### DISCUSSION TITLE
Can i alter clustered by column

### DISCUSSION THREAD

**Discussion by aman00323** [crate/crate#12121] [Q&A] [answered]
Is there any way to modify clustered by column? or if you can suggest a better way to update the column of clustered by without moving data?

**Comment by mfussenegger** [comment, marked as answer]
Altering the clustered by column is not possible. It is used for the data distrubtion and changing it would mean that they'd have to redistribute all the data.

Currently the best alternative is to create a new table, copy over the data using insert into with a select and then swap around the table with `alter cluster swap table`
